sympy/physics/control/lti.py

```diff
@@ -0,0 +1,1033 @@
+        [0.888105982187623]]), Matrix([
+        [1.81065556732437],
+        [1.50946135541217]])]
+
+        References
+        ==========
+
+        Joao P. Hespanha, Linear Systems Theory. 2009.
+        """
+        # verifiing valid arguments:
+        valid_flags = ('simplify', 'do_integrals')
+        for x in flags:
+            if x not in valid_flags:
+                raise ValueError('Unknown keyword argument: %s' % x)
+
+        try:
+            # assert right shape of u
+            if not u.shape[1] == 1:
+                raise ShapeError("u must not have more that one column, but has shape", u.shape)
+
+            if not self.represent[3].shape[1] == u.shape[0]:
+                raise ShapeError("u must have length", self.represent[3].shape[1])
+
+            # assert right shape of x0
+            if not x0.shape[1] == 1:
+                raise ShapeError("x0 must not have more than one column, but has shape", x0.shape)
+
+            if not self.represent[0].shape[1] == x0.shape[0]:
+                raise ShapeError("x0 must have length", self.represent[0].shape[1])
+
+        #
+        # Error handling
+        #
+        # if .shape goes wrong, a AttributeError is thrown
+        except AttributeError:
+            raise TypeError("u and x0 must be matrices!")
+
+        #
+        # find out if t is symbol, tuple or given wrong and call subroutines accordingly to that
+        #
+        sol = None
+
+        try:
+
+            # if t symbol, then calculate the solution symbolically
+            if isinstance(t, Symbol):
+                sol = self._solve_symbolically(u, x0, t, t0,
+                                               do_integrals=flags.get('do_integrals', True)
+                                               )
+                if flags.get('simplify', True):
+                    sol = simplify(sol)
+
+            # if not, try if it is tuple, list or sth.
+            elif isinstance(t[0], Symbol):
+                # if t[1] is a direct subclass of tuple or list
+                if isinstance(t[1], (list, tuple)):
+
+                    # use the private member function of the class to compute the numerical result
+                    sol = self._solve_numerically(u, x0, t[0], t[1], t0)
+
+                #  if its not, try to convert it
+                else:
+                    sol = self._solve_numerically(u, x0, t[0], list(t[1]), t0)
+
+        #
+        # Error handling
+        #
+        # index error can occur if t is not list-like
+        except IndexError:
+                IndexError("t must be symbol or have at least 2 entries")
+
+        # if the conversion goes wrong, its (hopefully) a TypeError
+        except TypeError:
+                TypeError("t[1] must be list, or list(t[1]) must work")
+
+        #
+        # if that worked, return the prestored solution
+        #
+
+        return sol
+
+    #
+    # _solve_numerically
+    #
+    def _solve_numerically(self, u, x0, t, t_list, t0):
```

Comment:
I do not think we should support solving ODEs numerically in this library. We should rely on lamdify and the code generation tools to produce functions that utlize numeric librarys (numpy, scipy, etc.) to find the numerical solutions to the ODEs. The only exception should be that we solve ODEs using desolve and then users can `evalf()` that solution to get numerical results.

Reply:
We discussed in #17866 that "Numeric solutions to the system should be done with python-control in general." So, I was planning to remove numerical evaluation. I'll try to define `solve()` for both classes that would produce symbolic solutions.